### ISSUE TITLE
Support openapi:// scheme for /#/ notation

### ISSUE THREAD

**Issue by devjack** [devjack/endpointer#4]
/#/ is a valid URI reference (for both file:// and http://

Instead, endpointer could support a custom scheme openapi:// that defines how to resolve `/#/` and other special notation within OpenAPI spec's

**Comment by MikeRalphson** [comment]
I think `/#/` resolves to `true` given the following object:

```js
{
    '#': {
      '': true
    }
}
```

What do you think about a `parent://` or `referrer://` scheme to access the referring document, and/or a `root://` scheme for accessing the root document?

**Comment by devjack** [comment]
As I follow this in the context of open API spec, `{$ref: "/#/"}` would be a JSON reference which is a URI; so '#' is, therefore, the fragment separator. After resolving the path component '/' (and assuming it somehow resolved) it would then proceed to resolve the JSON pointer in the fragment which is '/' to return the subdocument pointed to by '/`.

So JSON pointer `/#/` would  reference {"#":"foo"} but JSON Reference `/#/` would attempt to load the file '/'.

... I think ...

To the point of custom schemes, yes quite possibly. `referrer://` would have specificity concerns for how many documents reference others. `root://` is quite nice - I need to think on it...

**Comment by MikeRalphson** [comment]
See also https://github.com/whitlockjc/path-loader/issues/1 and `swagger2openapi`'s new [protocol/scheme handlers](https://github.com/Mermade/swagger2openapi/blob/master/docs/handlers.md#protocolscheme-handlers).

**Comment by handrews** [comment]
@MikeRalphson:

> I think /#/ resolves to true given the following object:
> 
```JSON
{
   "#": {
     "": true
   }
}
```

Not quite.  `#` is the fragment delimiter, so the actual JSON pointer here is `/`, which would resolve to `true` given this instance:

```JSON
{
    "": true
}
```

**Comment by handrews** [comment]
@devjack yes, the file `/` would be loaded, but then the pointer evaluated as in my last comment.

**Comment by devjack** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Support openapi:// scheme](https://trello.com/c/dowrjTtl/1-support-openapi-scheme)
